### PR TITLE
Add support for tf.distribute after enabling the update of cluster indices

### PR DESCRIPTION
**Pull request by Ruomei**
This PR adds the support for `tf.distribute` after enabling the update of cluster indices. Since these variables are deterministic (they depend on weights but not batch inputs), they are set as SyncOnRead so that they are calculated in each replica independently. In the model exporting time, they will be read from the first replica. Hope in this way we can avoid unnecessary sync for large LUTs.

This PR should follow https://github.com/tensorflow/model-optimization/pull/519, and it tests the change in cluster indices in cross-replica context (in the unit test). That is why it contains two commits.

### CODE REVIEW

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_distributed_test.py
Comment (akarmi) [inline tensorflow_model_optimization/python/core/clustering/keras/cluster_distributed_test.py:32]: Please, leave in only the strategies we want to be testing here, and remove unnecessary commented code here and in other places below.

**Review by akarmi — CHANGES_REQUESTED** [review]
Thank you. Just a nit from me. @wwwind, could you please have a look at this change as well?

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py
Comment (wwwind) [inline tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py:259]: the character '/' could be inside of the string

**Review by wwwind — APPROVED** [review]
looks good

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_distributed_test.py
Comment (Ruomei) [reply tensorflow_model_optimization/python/core/clustering/keras/cluster_distributed_test.py]: Thanks, done.

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py
Comment (Ruomei) [reply tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py]: Thanks, done.

**Comment by akarmi** [comment]
Thank you. Just waiting for #519 to land first, before merging this.

**Review by akarmi — APPROVED** [review]

**Comment by alanchiao** [comment]
Gotten https://github.com/tensorflow/model-optimization/pull/519 merged - apologize for the delay. If you're able to address the merge conflict, we can get this merged promptly.

**Comment by Ruomei** [comment]
No problem.
Thanks for merging #519. I closed it.
Just pushed the updated patch for this PR #531.

**Pull request merged**
